h2o-core/src/test/java/hex/RegexTokenizerTest.java

```diff
@@ -0,0 +1,65 @@
+package hex;
+
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import water.DKV;
+import water.Key;
+import water.ScopeTracker;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.fvec.Vec;
+
+import static org.junit.Assert.*;
+
+public class RegexTokenizerTest extends TestUtil  {
+
+  @Rule
+  public ScopeTracker _tracker = new ScopeTracker();
+
+  @BeforeClass
+  public static void stall() {
+    stall_till_cloudsize(1);
+  }
+
+  @Test
+  public void testTokenizeWithNAs() {
+    Frame f = new Frame(Key.<Frame>make());
+    f.add("C1", svec("a b c", null, "1 2", "cowabunga"));
+    f.add("C1", svec(null, null, "A", null));
+    DKV.put(f);
+    _tracker.track(f);
+
+    Frame tokenized = new RegexTokenizer(" ").transform(f);
+    _tracker.track(tokenized);
+
+    Vec expected = svec("a", "b", "c", null, null, "1", "2", "A", null, "cowabunga", null);
```

Comment:
WDYT about ignoring empty rows:
```svec("a", "b", "c", null, "1", "2", "A", null, "cowabunga", null)```

Reply:
If I ignore the empty rows I lose the mapping back to original sentences. Currently, I can rely on this invariant: the number of NAs in the output is equal to the number of rows. That lets me map the words back using a simple transformation (that H2O is capable of doing).

If I start skipping empty rows I might be not able to map it back so easily (because there is also filtering on minimum lenght - output might be empty even if the source had words).

Reply:
It makes sense. Thanks for your explanation!